js/join.js

```diff
@@ -0,0 +1,174 @@
+
+  //이메일의 내용이 비어있을 경우
+  if (input === "") {
+    showErrorMessage("이메일을 입력해주세요.", loginEmail, emailAlert);
+    loginBtnUpdate();
+    return;
+  }
+
+  // 이메일 형식이 맞지 않은 경우
+  if (!emailPattern.test(input)) {
+    showErrorMessage("잘못된 이메일 형식입니다.", loginEmail, emailAlert);
+    loginBtnUpdate();
+    return;
+  }
+
+  // 이메일이 정상일 경우 원래대로 바꾸기
+  hiddenErrorMessage(loginEmail, emailAlert);
+  loginBtnUpdate();
+}
+
+function passwordAlertFn(event) {
+  const input = event.target.value;
+  // 패스워드가 없을 경우
+  if (input === "") {
+    showErrorMessage("비밀번호를 입력해주세요.", loginPassword, passwordAlert);
+    loginBtnUpdate();
+    return;
+  }
+  // 패스워드가 8자보다 작은 경우우
+  if (input.length < 8) {
+    showErrorMessage(
+      "비밀번호를 8자 이상 입력해주세요.",
+      loginPassword,
+      passwordAlert
+    );
+    loginBtnUpdate();
+    return;
+  }
+  // 비밀번호가 정상일 경우 원래대로 바꾸기
+  hiddenErrorMessage(loginPassword, passwordAlert);
+  loginBtnUpdate();
+}
+
+function nicknameAlertFn(event) {
+  const input = event.target.value;
+  //닉네임이 빈 경우
+  if (input === "") {
+    showErrorMessage("닉네임을 입력해주세요.", joinNickname, nicknameAlert);
+    loginBtnUpdate();
+    return;
+  }
+
+  hiddenErrorMessage(joinNickname, nicknameAlert);
+  loginBtnUpdate();
+}
+
+function password2AlertFn(event) {
+  const input = event.target.value;
+  //비밀번호가 일치하지 않는 경우
+  if (input.length < 8) {
+    showErrorMessage(
+      "비밀번호를 8자 이상 입력해주세요.",
+      joinPassword2,
+      password2Alert
+    );
+    loginBtnUpdate();
+    return;
+  }
+  if (loginPassword.value !== joinPassword2.value) {
+    showErrorMessage(
+      "비밀번호가 일치하지 않습니다.",
+      joinPassword2,
+      password2Alert
+    );
+    loginBtnUpdate();
+    return;
+  }
+  hiddenErrorMessage(joinPassword2, password2Alert);
+  loginBtnUpdate();
+}
+
+function handleJoin(event) {
+  event.preventDefault();
+  window.location.href = "/pages/login.html";
+}
+
+function handleEyeIcon(event, Input, Icon) {
+  event.preventDefault();
+  if (Input.type === "password") {
+    Input.type = "text";
+    Icon.src = "../img/eye.png";
+  } else {
+    Input.type = "password";
+    Icon.src = "../img/no-eye.png";
+  }
+}
+
+// function handleEyeIcon2(event) {
+//   event.preventDefault();
+//   if (joinPassword2.type === "password") {
+//     joinPassword2.type = "text";
+//     eyeIcon2.src = "../img/eye.png";
+//   } else {
+//     joinPassword2.type = "password";
+//     eyeIcon2.src = "../img/no-eye.png";
+//   }
+// }
+
+loginEmail.addEventListener("focusout", emailAlertFn);
+loginPassword.addEventListener("focusout", passwordAlertFn);
+joinNickname.addEventListener("focusout", nicknameAlertFn);
+joinPassword2.addEventListener("focusout", password2AlertFn);
+joinBtn.addEventListener("click", handleJoin);
+toggleBtn.addEventListener("click", (event) =>
+  handleEyeIcon(event, loginPassword, eyeIcon)
+);
+toggleBtn2.addEventListener("click", (event) =>
+  handleEyeIcon(event, joinPassword2, eyeIcon2)
+);
```

Comment:
💊 제안
인자로 받는 것을 줄이고 싶다면 아래처럼도 가능합니다.
변경하라는 피드백보다 고민하는 것에 대한 예시라고 생각해주세요~

```suggestion
toggleBtn2.addEventListener("click", (event) =>
  event.preventDefault();
  
  handleEyeIcon(joinPassword2, eyeIcon2)
);
```